kube_syncer/secret_watcher.go

```diff
@@ -24,7 +24,7 @@ func (w SecretWatcher) Watch() <-chan abstractions.ISynchronizable {
 	secretsChan := make(chan abstractions.ISynchronizable)
 
 	go func() {
-		watcher, _ := w.kubeClient.CoreV1().Secrets(metav1.NamespaceAll).Watch(context.Background(), metav1.ListOptions{
+		watcher, _ := w.kubeClient.CoreV1().Secrets(metav1.NamespaceAll).Watch(context.TODO(), metav1.ListOptions{
```

Comment:
Background returns a non-nil, empty Context. It is never canceled, has no values, and has no deadline. It is typically used by the main function, initialization, and tests, and as the top-level Context for incoming requests.